tools/vm-builder/main.go

```diff
@@ -247,21 +247,41 @@ func main() {
 	}
 	defer cli.Close()
 
-	// pull source image
-	log.Printf("Pull source docker image: %s\n", *srcImage)
-	pull, err := cli.ImagePull(ctx, *srcImage, types.ImagePullOptions{})
+	hostContainsSrcImage := false
+	hostImages, err := cli.ImageList(ctx, types.ImageListOptions{})
 	if err != nil {
 		log.Fatalln(err)
 	}
-	defer pull.Close()
-
-	// do quiet pull - discard output
-	io.Copy(io.Discard, pull)
-	/*
-	   if err = printReader(pull); err != nil {
-	   	log.Fatalln(err)
-	   }
-	*/
+
+	for _, img := range hostImages {
+		for _, name := range img.RepoTags {
+			if name == *srcImage {
```

Comment:
one of the things I hate the most about go's flag package is that it defaults to returning pointers to fundametnal types in ways that you never want to deal with, and that means you end up having to derefrence variables in awkward places. Reading this code gives me the whillies because `srcImage` could be nil so this would panic, and if it's not, what would otherwise be a stack variable is on the heap. If this weren't running interactively or in build scripts that would be bad, but given that this is essentially a build script, it's fine. 

About CLI option parsing: When I find myself needing to use the stdlib flags library, I: 
- declare variables and then use the `p` flavored functions and pass pointers 
- attempt to avoid putting as much of this into the global (ok package) scope as possible, mostly by putting things in functions somewhere. 

Most of the time, for bigger options, people use a CLI wrapper like (my favorite) `github.com/urfave/cil` or (more popularly,) `github.com/spf13/cobra`, but these are both overkill for things like this.